### PR TITLE
SUPINT-1912: add docker image for common project

### PR DESCRIPTION
**Pull request by atchertchian**

### CODE REVIEW

File: ci/Jenkinsfiles/build.groovy [hunk 1]
```diff
@@ -119,6 +161,171 @@ pipeline {
...
+          Registry: ${DOCKER_REGISTRY}
+          Nuxeo Image tag: ${NUXEO_IMAGE_VERSION}
+          """
+          script {
+            // build and push Docker images to the Jenkins X internal Docker registry
+            def dockerPath = 'docker'
+            sh "envsubst < ${dockerPath}/skaffold.yaml > ${dockerPath}/skaffold.yaml~gen"
+            sh """#!/bin/bash +x
+              skaffold build -f ${dockerPath}/skaffold.yaml~gen
+            """
```
Comment (ataillefer) [inline ci/Jenkinsfiles/build.groovy]:
```suggestion
            sh "skaffold build -f ${dockerPath}/skaffold.yaml~gen"""
```
single line => `"`, no credential to hide, no nee for `bash +x` to not trace the commands

File: ci/Jenkinsfiles/build.groovy [hunk 2]
```diff
@@ -119,6 +161,171 @@ pipeline {
...
+          ----------------------------------------
+          Deploy preview environment
+          ----------------------------------------"""
+          dir('helm/preview') {
+
+            script {
+              // first substitute docker image names and versions
+              sh """
+                cp values.yaml values.yaml.tosubst
+                DOCKER_IMAGE_NAME=${DOCKER_IMAGE_NAME} envsubst < values.yaml.tosubst > values.yaml
```
Comment (ataillefer) [inline ci/Jenkinsfiles/build.groovy]:
```suggestion
                envsubst < values.yaml.tosubst > values.yaml
```
as `DOCKER_IMAGE_NAME` is already in the environment

File: ci/Jenkinsfiles/build.groovy [hunk 3]
```diff
@@ -53,6 +85,16 @@ pipeline {
     REFERENCE_BRANCH = 'master'
     SCM_REF = "${getCommitSha1()}"
     VERSION = "${getVersion(REFERENCE_BRANCH)}"
+
+    PERSISTENCE = "${BRANCH_NAME == REFERENCE_BRANCH}"
```
Comment (ataillefer) [inline ci/Jenkinsfiles/build.groovy]: seems unused

File: ci/Jenkinsfiles/build.groovy [hunk 4]
```diff
@@ -119,6 +161,171 @@ pipeline {
...
+
+              // second create target namespace (if doesn't exist) and copy secrets to target namespace
+              String currentNs = sh(returnStdout: true, script: 'jx -b ns | sed -r "s/^Using namespace \'([^\']+)\'.+\\$/\\1/"').trim()
+              boolean nsExists = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+              // Only used with jx preview on pr branches
+              String noCommentOpt = '';
+              if (nsExists) {
+                noCommentOpt = '--no-comment'
+                // Previous preview deployment needs to be scaled to 0 to be replaced correctly
+                sh "kubectl --namespace ${PREVIEW_NAMESPACE} scale deployment preview --replicas=0"
```
Comment (ataillefer) [inline ci/Jenkinsfiles/build.groovy]: no MongoDB/ES with persistence, shouldn't be necessary

File: ci/Jenkinsfiles/build.groovy (hunk 3 above)
Comment (atchertchian) [reply ci/Jenkinsfiles/build.groovy]: This is a remnant of the persistent preview indeed, could be cleaned up

**Comment by atchertchian** [comment]
Back to draft to experiment with GH action --> will see if pipeline preview is still useful after that

File: ci/Jenkinsfiles/build.groovy [hunk 5]
```diff
@@ -119,6 +160,172 @@ pipeline {
...
+          """
+          script {
+            // build and push Docker images to the Jenkins X internal Docker registry
+            def dockerPath = 'docker'
+            sh "envsubst < ${dockerPath}/skaffold.yaml > ${dockerPath}/skaffold.yaml~gen"
+            withCredentials([string(credentialsId: 'instance-clid', variable: 'INSTANCE_CLID')]) {
+
+              // replace lines by "--"
+              String clid = sh(returnStdout: true, script: '''#!/bin/bash +x
+              echo -e "${INSTANCE_CLID}" | sed ':a;N;\$!ba;s/\\n/--/g'
```
Comment (BoboTiG) [inline ci/Jenkinsfiles/build.groovy:189]: Wow :)

**Review by BoboTiG — APPROVED** [review]
It seems fine about GitHub/Jenkins stuff. I am not familiar with the overall thing, so do not take my review as a complete one :)

File: ci/Jenkinsfiles/build.groovy (hunk 5 above)
Comment (atchertchian) [reply ci/Jenkinsfiles/build.groovy:189]: Yeah well, we do have this in all pipelines (carriage return in the clid is not handled well... so it's replaced).

Thanks for the review: Antoine already had a look at the docker/Jenkins part so we should be good :)

**Pull request merged**